src/commands/tcam/list-apis.ts

```diff
@@ -3,75 +3,82 @@
  * This file is subject to the license terms contained
  * in the license file that is distributed with this file.
  */
-import {flags} from '@oclif/command';
+import { flags } from '@oclif/command';
 import { TCBaseCommand, ux } from '@tibco-software/cic-cli-core';
-import { CLIAPIS } from '../../constants/url.constants';
+import { CLIAPIS } from '../../utils/url.constants';
+
 export default class TcamListApis extends TCBaseCommand {
-  static description = 'Lists APIs';
-  static examples: string[] | undefined = ["tibco tcam:list-apis",
-  "tibco tcam:list-apis -p 'Cli Project'","tibco tcam:list-apis -p 'Cli Project,AuthProject'",
-   "tibco tcam:list-apis -t 'openapi'","tibco tcam:list-apis -p 'AuthProject' -t 'openapi'",
-   "tibco tcam:list-apis -a 'CliAsyncApi,CliOpenApi'"
-   ];
-  spinner:any;
+  static description = 'List APIs';
+  static examples: string[] | undefined = [`tibco tcam:list-apis`,
+    `tibco tcam:list-apis --projectnames "Cli Project"`, `tibco tcam:list-apis --projectnames "Cli Project,AuthProject"`,
+    `tibco tcam:list-apis --apitypes "openapi"`, `tibco tcam:list-apis -p "AuthProject" -t "openapi"`,
+    `tibco tcam:list-apis --apinames "CliAsyncApi,CliOpenApi"`
+  ];
+  spinner: any;
   static flags: flags.Input<any> & typeof TCBaseCommand.flags = {
     ...TCBaseCommand.flags,
-    help: flags.help({char: 'h'}),
+    help: flags.help({ char: 'h' }),
     // flag with no value (-f, --force)
-    force: flags.boolean({char: 'f'}),
-    projectname: flags.string({char: 'p', description: 'Project names'}),
-    apitypes: flags.string({char: 't', description: 'API types you want to list. For example openapi,asyncapi'}),
-    apinames: flags.string({char: 'a', description: 'API names'})
+    force: flags.boolean({ char: 'f' }),
+    projectnames: flags.string({ char: 'p', description: 'Specify project names' }),
+    apitypes: flags.string({ char: 't', description: 'API types you want to list. For example openapi,asyncapi' }),
+    apinames: flags.string({ char: 'a', description: 'Specify API names' })
   }
+
   async init() {
     await super.init();
     // Do any other  initialization
-   this.spinner = await ux.spinner();
+    this.spinner = await ux.spinner();
   }
+
   async run() {
-    const {flags} = this.parse(TcamListApis);
+    const { flags } = this.parse(TcamListApis);
     let params = new URLSearchParams();
-    if(flags.projectname){
-     const  projects = flags.projectname.split(',').map((name:any) => name.trim());
-    projects.forEach((proj:string) => {
-      params.append('projectNames',proj);
-    });
+    if (flags.projectnames) {
+      const projects = flags.projectnames.split(',').map((name: any) => name.trim());
+      projects.forEach((proj: string) => {
+        params.append('projectNames', proj);
```

Comment:
this logic is repeated 2 more times for the other 2 flags. Consider extracting the logic to a reusable function.
I'm thinking something like this for a simple change:
```ts
parseListItems(flags.apitypes)
	.forEach((type) => params.append('apiTypes', type))
```

Or encapsulate all in a function and pass the params as parameter:

```ts
if (flag.apitypes) {
  addListParams(params, flags.apitypes, 'apiTypes');
}

//
function addListParams(params: URLSearchParams, flagListValue: string, paramName: string) {
  flagListValue
    .split(',')
    .forEach(item => {
       params.push(paramName, item.trim())
    })
    
}
```

Reply:
A common function called **_addListParams_** added. 
Coomit link: https://github.com/TIBCOSoftware/cic-cli-plugin-tcam/commit/df0e8675f2c45bf102d0a2099add907ce064677b